_posts/2020-03-30-exploting-spatial-diversity-using-unetstack3.md

```diff
@@ -0,0 +1,98 @@
+wormhole.dsp = 'udp'                          (3)
+```
+Line number (1) adds a [`UdpLink`](https://unetstack.net/handbook/unet-handbook_wired_and_over_the_air_links.html) agent that implements a link protocol over UDP/IP for use over wired/wireless IP networks. The next step is to add the `Wormhole` agent and make it utilize the infrastructure provided by the `UdpLink` agent to allow the agents residing on receivers nodes to transparently talk to each other! 
+
+Line number (2) adds a `Wormhole` agent which allows the [fjåge](https://fjage.readthedocs.io/en/latest/) messages to be sent between containers over a [Unet link](https://unetstack.net/handbook/unet-handbook_introduction.html). 
+
+Line number (3) tells the `Wormhole` agent to utilize the `UdpLink` as the Unet link to share information. For more details on the `Wormhole` agent, please refer to this [link](https://unetstack.net/handbook/unet-handbook_preface.html).
+
+*Additional set up on the assisting receiver* :
+
+We may not want to share the information from all agents, instead, we are interested in messages that are published on the Physical agent's topic on the main receiver.
+```groovy
+wormhole.broadcast = [topic(phy), topic(phy, Physical.SNOOP)]
+```
+The above line of code is added only on the assisting receivers to forward only the messages received on the `Physical` agent's topic to be sent over the `Wormhole` link.
+
+#### 2. Add the `Unity` agent on the main receiver:
+
+Now that the receiver nodes are ready to cooperate, we can go ahead and add the `Unity` agent on the main receiver node. 
+
+```groovy
+container.add 'unity', new Unity()
+```
+If a user wishes to take a look at the parameters of the `Unity` agent after loading the agent, type `unity` on the node's web shell interface:
+```
+> unity
+« Spatial diversity agent »
+
+Combines copies of information received from multiple 
+receivers to decode packets.
+
+[org.arl.unet.unity.UnityParam]
+  assisters = []
+  enable = false
+  maxAssisterRange = 100.0
+  phy = phy
+```
+The `assisters` parameter is used to store the node addresses of the assisting receivers that are assisting the main receiver. The user might also want to change the default value of `maxAssisterRange` if the furthest assisting receiver to the main receiver is at a distance larger than 100 m.
+
+```groovy
+unity.assisters = [< first_assister_address >, <second_assister_address>]
+assister.enable = true
+```
+Once the assisters node addresses are set, as shown above, the `Unity` agent is enabled. And voila, you are ready to see the benefits of cooperating receivers in terms of reliability and effective data rate. 
+
+
+**NOTE**: We will be sharing white papers and material detailing the protocol implemented once it is available. Until then, for keen readers and folks interested in knowing more details on learning and using this technology, please get in touch by emailing at [info@subnero.com](https://subnero.com/contact/). 
```

Comment:
"they", not "it". But I find this note awkward. Get feedback from others and decide.

Reply:
Removed this as I found it odd too in the second reading. Add much more details on the expected results instead.